cmd/src/server.rs

```diff
@@ -147,6 +156,60 @@ impl TiKVServer {
         }
     }
 
+    /// Initialize and check the config
+    ///
+    /// Warnings are logged and fatal errors exist.
+    ///
+    /// #  Fatal errors
+    ///
+    /// - If `dynamic config` feature is enabled and failed to register config to PD
+    /// - If some critical configs (like data dir) are differrent from last run
+    /// - If the config can't pass `validate()`
+    /// - If the max open file descriptor limit is not high enough to support
+    ///   the main database and the raft database.
+    fn init_config(mut config: TiKvConfig, pd_client: Arc<RpcClient>) -> ConfigController {
+        let mut version = Default::default();
+        if config.dynamic_config {
+            // Server address and cluster id can not be changed
+            let addr = config.server.addr.clone();
+            let cluster_id = config.server.cluster_id;
+            // Using the same file for initialize global logger
+            // and diagnostics service
+            let log_file = config.log_file.clone();
+
+            let (v, mut cfg) = ConfigHandler::create(addr.clone(), Arc::clone(&pd_client), config)
+                .unwrap_or_else(|e| fatal!("failed to register config from pd: {}", e));
+
+            cfg.server.addr = addr;
+            cfg.server.cluster_id = cluster_id;
+            cfg.log_file = log_file;
+            cfg.dynamic_config = true;
+            config = cfg;
+            version = v;
+        }
+
+        validate_and_persist_config(&mut config, true);
```

Comment:
Do we still need to do such thing later?

Reply:
Yes, we still need to, because config returned from PD have not validate.